Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -125,16 +125,16 @@ Element* ScrollTimeline::source() const
     case Scroller::Nearest: {
         if (CheckedPtr subjectRenderer = source->renderer()) {
             if (CheckedPtr nearestScrollableContainer = subjectRenderer->enclosingScrollableContainer()) {
-                if (RefPtr nearestSource = nearestScrollableContainer->element()) {
-                    Ref document = nearestSource->document();
+                if (auto* nearestSource = nearestScrollableContainer->element()) {
+                    Ref document = Ref { *nearestSource }->document();
                     RefPtr documentElement = document->documentElement();
                     if (nearestSource != documentElement)
```

Comment:
If I understood @geoffreygaren correctly this is not safe as `nearestSource` is a raw pointer.

Reply:
Yes, I agree with @geoffreygaren . The `if (nearestSource != documentElement)` is not safe if `nearestSource` isn't kept alive, due to pointer reuse.